ios/Codeforces Watcher/Application/features/problems/ProblemsTableViewAdapter.swift

```diff
@@ -13,43 +13,48 @@ import common
 class ProblemsTableViewAdapter: NSObject, UITableViewDelegate, UITableViewDataSource {
     
     var problems: [Problem] = []
+    var visibleProblems: [Problem] = []
 
     var onProblemClick: ((String, String) -> ())?
 
     func numberOfSections(in tableView: UITableView) -> Int {
         return 1
     }
+    
+    var isFavourite = false
```

Comment:
It's better to set string of "empty problems" explanation, than boolean here. (So adapter is as dummy as possible).